version: '3.5'
services:
  postgres:
    container_name: postgres
    image: postgres:latest
    restart: on-failure
    environment:
      POSTGRES_PASSWORD: Rand0m!
      POSTGRES_USER: root
    ports:
      - "8001:5432"
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./docker/postgres/init.sql:/docker-entrypoint-initdb.d/init.sql

  pgadmin:
    container_name: pdadmin
    image: dpage/pgadmin4:latest
    restart: on-failure
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: Rand0m!
      POSTGRES_DB: postgres
    ports:
      - "8002:80"
    volumes:
      - pgadmin-data:/var/lib/pgadmin

  app:
    container_name: app
    build:
      context: .
    restart: on-failure
    ports:
      - "8003:8080"
    tty: true
    depends_on:
      - postgres
    links:
      - postgres

  swagger-ui:
    image: swaggerapi/swagger-ui
    environment:
      - API_URL=http://localhost:8003/api/swagger
    ports:
      - 9000:8080

volumes:
  postgres-data:
    external: true
  pgadmin-data:
    external: true